service/src/test/java/com/epam/rft/atsy/service/impl/CandidateServiceImplTest.java

```diff
@@ -120,6 +132,46 @@ public void getCandidateShouldReturnProperDtoWhenIdExists() {
   }
 
   @Test(expected = IllegalArgumentException.class)
+  public void getCandidateByApplicationIDShouldThrowIllegalArgumentExceptionWhenApplicationIDIsNull() {
+    // When
+    candidateService.getCandidateByApplicationID(null);
+  }
+
+  @Test
+  public void getCandidateByApplicationIdShouldReturnNullWhenThereIsNoApplicationWithTheGivenId() {
+    // Given
+    given(applicationsRepository.findOne(APPLICATION_ID))
```

Comment:
The given situation described in this line will never executed by the production code, as the `findOne` will be called using `NON_EXISTENT_APPLICATION_ID` parameter.

As there is no output defined for non existent application id, the `findOne` will return null by default.

Please make it clear that the findOne method gets called with the `NON_EXISTENT_APPLICATION_ID` and explicitly show that it will return null.